src/V3Ast.cpp

```diff
@@ -1151,7 +1153,7 @@ void AstNode::dumpTreeFile(const string& filename, bool append, bool doDump, boo
             if (logsp->fail()) v3fatal("Can't write " << filename);
             *logsp << "Verilator Tree Dump (format 0x3900) from <e" << std::dec << editCountLast();
             *logsp << "> to <e" << std::dec << editCountGbl() << ">\n";
-            if (editCountGbl() == editCountLast() && !(v3Global.opt.dumpTree() >= 9)) {
+            if (editCountGbl() == editCountLast() && ::dumpTree() < 9) {
```

Comment:
> The ! >= was so that when I grep for debug I can make sure >= is always used.

Ah, that was the reason why I was considering dumpTree(9) (which is the current dumpTree() >= 9).

There is no `opt.dumpTree()`, there is only `opt.dumpLevel("tree")`. `dumpTree()` is defined in the global namespace via VL_DEFINE_DEBUG_FUNCTIONS, and `::dumpTree()` picks up that. The `::` in this one instance is necessary because this is AstNode, which itself has an `AstNode::dumpTree(const string& = ""), which is what you would reference without the leading `::`. This is why I was considering renaming to `doDumpTree`.